### PR TITLE
[#1966] improvement(core): improve TreeLock perf using Deque

### PR DESCRIPTION
**Pull request by YxAc**
### What changes were proposed in this pull request?
- Use Deque instead of Stack to improve TreeLock perf
- Optimize the logic of unlock method for better readability

### Why are the changes needed?
Fix: #1966 

### Does this PR introduce _any_ user-facing change?
N/A

### How was this patch tested?
By hand

### CODE REVIEW

File: core/src/main/java/com/datastrato/gravitino/lock/TreeLock.java [hunk 1]
```diff
@@ -63,7 +64,7 @@ public class TreeLock {
   private final List<TreeLockNode> lockNodes;
 
   // TreeLockNode that has been locked.
-  private final Stack<TreeLockNode> heldLocks = new Stack<>();
+  private final Deque<TreeLockNode> heldLocks = new ArrayDeque<>();
```
Comment (yuqi1129) [inline core/src/main/java/com/datastrato/gravitino/lock/TreeLock.java:67]: @YxAc 
Thank you for bringing this to our attention. To improve it, I suggest we only need to keep the last 
Index the locked node and then unlock it from the index to 0.

**Comment by YxAc** [comment]
@yuqi1129 Can you help review again? thanks

File: core/src/main/java/com/datastrato/gravitino/lock/TreeLock.java (hunk 1 above)
Comment (YxAc) [reply core/src/main/java/com/datastrato/gravitino/lock/TreeLock.java:67]: > To improve it, I suggest we only need to keep the last
> Index the locked node and then unlock it from the index to 0.

As discussed offline, let's keep the stack for the time being, better readable.

**Review by yuqi1129 — APPROVED** [review]
LGTM

**Comment by jerryshao** [comment]
@YxAc thanks a lot for your contribution.

**Pull request merged**